### ISSUE TITLE
MTY_HttpRequest - Mac 

### ISSUE THREAD

**Issue by chadmortimer** [matoya/libmatoya#27]
MTY_HttpRequest doesn't seem to work correctly on Mac. Built lib from latest source 

The request itself works and returns true however the response return parameter is null and the status code is 0. The same request works fine on Windows returning the correct response data and code 200.

MTY_GetLog doesn't reveal any errors

linked -framework Security as this seemed to be a dependency 
MacBook Pro (15-inch, 2018)
Big Sur 11.2.3

**Comment by chrisd1100** [comment]
Can you try running the test suite on the same machine? It makes a number of http requests, I'm seeing good behavior on my mac. Also, would you mind sharing the arguments to the funcion?

**Comment by chadmortimer** [comment]
Ran the test and see a bunch of passes but I see the tests don't check the resp / resp code (only that MTY_HttpRequest returns true). So I modified the test to print code and response and I am seeing valid results (i.e response content and code 200)

I've tested only two requests so far, one with my dev api and then when that didn't work I tried the request in the examples (the one that retrieves a png) like so:

MTY_HttpRequest("user-images.githubusercontent.com", 0, true, "GET",
		"/328897/112402607-36d00780-8ce3-11eb-9707-d11bc6c73c59.png",
		NULL, NULL, 0, 5000, &png, &png_size, &code)
		
This didn't seem to work for me either. The request returns true but resp parameter is null and resp code is 0

....And as I am typing this I spotted that in the tests you're including -framework Foundation, I have added this to my list and now its working! There's obviously a dependancy there and yet somehow it doesn't produce any compilation or runtime errors without that framework?
